src/webview/mod.rs

```diff
@@ -58,7 +58,25 @@ pub use url::Url;
 use crate::application::platform::windows::WindowExtWindows;
 use crate::application::{dpi::PhysicalSize, window::Window};
 
-use http::{Request, Response};
+use http::{Request, Response as HttpResponse};
+
+/// Resolves a custom protocol [`Request`] asynchronously.
+///
+/// See [`WebViewBuilder::with_asynchronous_custom_protocol`] for more information.
+pub struct Response {
```

Comment:
I don't like this name, it conflicts with the re-exported `http::Response`, this should be renamed to `RequestAsyncResponder`

Reply:
I like this name, what do you think @wusyong

Reply:
Ok I can agree on `Result` can be removed.
But I think it becomes more and more complicated lol.
How about custom protocol attribute to be: 

```
custom_protocols: Vec<String, Box<dyn Fn(CustomURLHandler)>
```

And `CustomURLHandler` has two methods:
```
fn request(&self) -> &Request<Vec<u8>
fn response(self, Response<Cow<'static, [u8]>>>)
```

This also keep `use http::{request, response}`.

Reply:
The downside with this approach is I'd like to call `request.into_parts()` so I don't have to clone the request body or other data, and it needs to own the Request reference.

If we expose the request field itself it would be fine, but I still don't see much difference to the existing API.

Reply:
Original `with_custom_protocol` can still keep the signature (with Result removed.)
`F: Fn(&Request<Vec<u8>>) -> Response<Cow<'static, [u8]>> + 'static`

Reply:
For the original one i think it's fine to keep the Result (just to not have a breaking change for wry devs).

Reply:
@lucasfernog Well, just rename it to `AsyncResponder` then.
I still think this type will need a better name or more detailed docs.
Otherwise, it's going to be confusing for users.

Reply:
If we really don't want to confuse users, then rename it to `CustomProtocolAsyncResponder` :joy: and the docs seems sufficient enough for its use-case especially since it links back to the `with_async_custom_protocol`.

Reply:
whatever you guys decide

Reply:
Alright keep it this way.